federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPOperationManager.java

```diff
@@ -370,26 +370,36 @@ private SearchControls getSearchControls(Collection<String> returningAttributes,
     }
 
     public String getFilterById(String id) {
-        String filter = null;
+        StringBuilder filter = new StringBuilder();
+        filter.insert(0, "(&");
 
         if (this.config.isObjectGUID()) {
             byte[] objectGUID = LDAPUtil.encodeObjectGUID(id);
-
-            filter = "(&(objectClass=*)(" + getUuidAttributeName() + LDAPConstants.EQUAL + LDAPUtil.convertObjectGUIDToByteString(objectGUID) + "))";
+            filter.append("(objectClass=*)(").append(
+                    getUuidAttributeName()).append(LDAPConstants.EQUAL)
+                .append(LDAPUtil.convertObjectGUIDToByteString(
+                    objectGUID)).append(")");
 
         } else if (this.config.isEdirectoryGUID()) {
-            filter = "(&(objectClass=*)(" + getUuidAttributeName().toUpperCase() + LDAPConstants.EQUAL + LDAPUtil.convertGUIDToEdirectoryHexString(id) + "))";
+            filter.append("(objectClass=*)(").append(getUuidAttributeName().toUpperCase())
+                .append(LDAPConstants.EQUAL
+                ).append(LDAPUtil.convertGUIDToEdirectoryHexString(id)).append(")");
+        } else {
+            filter.append("(objectClass=*)(").append(getUuidAttributeName()).append(LDAPConstants.EQUAL)
+                .append(id).append(")");
         }
 
-        if (filter == null) {
-            filter = "(&(objectClass=*)(" + getUuidAttributeName() + LDAPConstants.EQUAL + id + "))";
+        if (config.getCustomUserSearchFilter() != null) {
+            filter.append(config.getCustomUserSearchFilter());
         }
 
         if (logger.isTraceEnabled()) {
-            logger.tracef("Using filter for lookup user by LDAP ID: %s", filter);
+            logger.tracef("Using filter for lookup user by LDAP ID: %s", filter.toString());
         }
```

Comment:
Also don't add the `toString`, this is done by the formatter in the logger anyway.
The correct form is without the enclosing `if (logger.isTraceEnabled())` as that is again done in the `tracef` anyway.
```suggestion
        logger.tracef("Using filter for lookup user by LDAP ID: %s", filter);
```

Reply:
@hmlnarik Good catch!

Reply:
@hmlnarik Thanks. I tried to make the code model closer to other classes that do similar things. At the end, if we are going to reuse the generated String I extract it straight to a variable, so we can make use of String.intern(). The indentation has been fixed @mposolda , thanks for the tip.